src/com/amazon/ion/IonReader.java

```diff
@@ -321,8 +321,7 @@ public interface IonReader
 
     /**
      * Returns the current value as a {@link BigInteger}.  This is only valid if there
-     * is an underlying value and the value is of a numeric type (int, float, or
-     * decimal).
+     * is an underlying value and the value is of type int or float.
```

Comment:
TBH, I'd also be in favor of getting rid of the conversion for `float` so that potentially lossy conversions must be performed explicitly by the consumer of this library. What do you think?

Reply:
Yes, I want to do a thorough audit of all of these conversions and eliminate all of them that don't seem to be used. #523